oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/query/IMetadataQueryDAO.java

```diff
@@ -20,58 +20,37 @@
 
 import java.io.IOException;
 import java.util.List;
-import org.apache.skywalking.oap.server.core.analysis.NodeType;
-import org.apache.skywalking.oap.server.core.query.type.Database;
 import org.apache.skywalking.oap.server.core.query.type.Endpoint;
 import org.apache.skywalking.oap.server.core.query.type.Service;
 import org.apache.skywalking.oap.server.core.query.type.ServiceInstance;
 import org.apache.skywalking.oap.server.core.storage.DAO;
 
 public interface IMetadataQueryDAO extends DAO {
     /**
-     * @param group group name for filtering.
-     * @return list of the all available normal services
+     * @param layer layer name for filtering
+     * @param group group name for filtering
+     * @return list of the all available services
      */
-    List<Service> getAllServices(final String group) throws IOException;
+    List<Service> listServices(final String layer, final String group) throws IOException;
 
-    /**
-     * @return list of the all available browser services
-     */
-    List<Service> getAllBrowserServices() throws IOException;
-
-    /**
-     * @return list of all conjecture database services.
-     */
-    List<Database> getAllDatabases() throws IOException;
+    Service findService(final String serviceId) throws IOException;
 
     /**
-     * @param nodeType describe which kind of node of Service
-     * @param keyword  to filter the normal service
-     * @return the list of normal services matching the given keyword
+     * @param startTimestamp The instance is required to be live after this timestamp
+     * @param endTimestamp   The instance is required to be live before this timestamp.
+     * @param serviceId      the owner of the instances.
+     * @return list of instances matching the given conditions.
      */
-    List<Service> searchServices(final NodeType nodeType, final String keyword) throws IOException;
+    List<ServiceInstance> listInstances(final long startTimestamp, final long endTimestamp,
+                                        final String serviceId) throws IOException;
 
-    /**
-     * @param nodeType    describe which kind of node of Service
-     * @param serviceCode to literal match
-     * @return the service matching the given full name.
-     */
-    Service searchService(final NodeType nodeType, final String serviceCode) throws IOException;
+    ServiceInstance getInstance(final String instanceId) throws IOException;
```

Comment:
Please add an E2E to verify this new API, from the implementation it seems this is not working

Reply:
@wankai123 Please recheck and do manual GraphQL testing. 

@kezhenxu94 e2e is not working at this stage I am afraid. CLI is lack of env to verify its feature. We have to use manual tests for some parts.

Reply:
Manual tested, it works fine. Once CLI supported the new query, I'll add the new verify.